src/main/java/org/jitsi/videobridge/BitrateController.java

```diff
@@ -0,0 +1,613 @@
+            }
+        }
+        else
+        {
+            ssrc = RTCPHeaderUtils.getSenderSSRC(buf, off, len);
+            if (ssrc < 0)
+            {
+                return false;
+            }
+
+        }
+
+        SimulcastController simulcastController
+            = ssrcToBitrateController.get((int) ssrc);
+
+        return simulcastController != null
+            && simulcastController.rtpTranslatorWillWrite(data, buf, off, len);
+    }
+
+    /**
+     * Computes a new bitrate allocation for every endpoint in the conference,
+     * and updates the state of this instance so that bitrate allocation is
+     * eventually met.
+     *
+     * @param conferenceEndpoints the ordered list of {@link Endpoint}s
+     * participating in the multipoint conference with the dominant (speaker)
+     * {@link Endpoint} at the beginning of the list i.e. the dominant speaker
+     * history. This parameter is optional but it can be used for performaance;
+     * if it's omitted it will be fetched from the
+     * {@link ConferenceSpeechActivity}.
+     */
+    public void update(List<Endpoint> conferenceEndpoints)
+    {
+        // Gather the conference allocation input.
+        if (conferenceEndpoints == null)
+        {
+            conferenceEndpoints
+                = dest.getConferenceSpeechActivity().getEndpoints();
+        }
+
+        long bweBps = Long.MAX_VALUE;
+
+        // Compute the bitrate allocation.
+        EndpointBitrateAllocation[]
+            allocations = allocate(bweBps, conferenceEndpoints);
+
+        List<String> newForwardedEndpoints = new ArrayList<>();
+
+        for (int i = 0; i < allocations.length; i++)
+        {
+            int ssrc = allocations[i].targetSSRC,
+                targetIdx = allocations[i].targetIdx;
+
+            SimulcastController ctrl = ssrcToBitrateController.get(ssrc);
+            if (ctrl == null && allocations[i].track != null)
+            {
+                ctrl = new SimulcastController(allocations[i].track);
+
+                RTPEncodingDesc[] rtpEncodings
+                    = allocations[i].track.getRTPEncodings();
+
+                // Route all encodings to the specified bitrate controller.
+                for (int j = 0; j < rtpEncodings.length; j++)
+                {
+                    ssrcToBitrateController.put(
+                        (int) rtpEncodings[j].getPrimarySSRC(), ctrl);
+
+                    if (rtpEncodings[j].getRTXSSRC() != -1)
+                    {
+                        ssrcToBitrateController.put(
+                            (int) rtpEncodings[j].getRTXSSRC(), ctrl);
+                    }
+                }
+            }
+
+            if (ctrl != null)
+            {
+                ctrl.update(targetIdx);
+            }
+
+            if (targetIdx > -1)
+            {
+                newForwardedEndpoints.add(allocations[i].endpointID);
+            }
+        }
+
+        this.forwardedEndpoints = newForwardedEndpoints;
```

Comment:
Shouldn't wewe notify the receiver that the forwardedEndpoints changed (if they did)?